### ISSUE TITLE
Ymlr produces invalid yaml for maps containing key with special chars

### ISSUE THREAD

**Issue by elliottneilclark** [ufirstgroup/ymlr#94]
#### Issue description

Take the following example:

```elixir
 %{ test: %{"@type" => 100}} |> Ymlr.Encoder.to_s! 
```

That should produce:

```
test:
  "@type": 100
```

Instead it produces:
```
test:
  @type: 100
```
Likely this encode to io list for map will need to be changed here: https://github.com/ufirstgroup/ymlr/blob/main/lib/ymlr/encoder.ex#L86

#### Link to documentation

According to the [spec](https://yaml.org/spec/1.2.2/#rule-c-directive) `@` is reserved and should be quoted in the beginning of all strings.

**Comment by jlgeering** [comment]
applies both to @ and `, in at the start of keys and values => at least 4 new test cases

**Comment by jlgeering** [comment]
work in progress branch (only failing tests) #95

**Comment by jlgeering** [comment]
actually we already handle reserved indicators in values, the problem is that we didn't quote map keys when needed

first round of improvement in #95

**Comment by jlgeering** [comment]
future work: we should probably throw an error if map keys contains newlines (or use double quotes, not sure what the right thing to do is)

...

**Comment by jlgeering** [comment]
changed my mind: newlines seem to be allowed in map keys, they just need to be encoded properly

**Comment by jlgeering** [comment]
I believe this was released with v3.0.1, see https://github.com/ufirstgroup/ymlr/releases/tag/v3.0.1
